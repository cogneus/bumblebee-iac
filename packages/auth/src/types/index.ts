export { TokenResponse } from './token-response.interface'
export { TokenServiceAttrs } from './token-service-attrs.interface'
