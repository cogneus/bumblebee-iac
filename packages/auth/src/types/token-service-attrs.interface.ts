export interface TokenServiceAttrs {
  issuer: string
  audience: string
  expiresIn: string | number
}
