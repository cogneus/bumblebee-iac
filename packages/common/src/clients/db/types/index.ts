export * from './db-client.interface'
export * from './paged-items.interface'
export * from './db-client-options.interface'
export * from './table-config.interface'
export * from './transact-item.interface'
