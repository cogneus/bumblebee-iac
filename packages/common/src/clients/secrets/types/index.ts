export * from './secrets-client.interface'
export * from './secrets-client-options.interface'
