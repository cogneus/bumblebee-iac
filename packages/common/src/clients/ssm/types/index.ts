export * from './sm-client.interface'
export * from './sm-client-options.interface'
