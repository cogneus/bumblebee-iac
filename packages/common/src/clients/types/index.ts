export * from './destroyable.interface'
export * from './client-options.interface'
export * from './clients.interface'
