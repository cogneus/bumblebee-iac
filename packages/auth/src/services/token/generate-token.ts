import { sign } from 'jsonwebtoken'
import { Token } from 'bumblebee-common'
import { TokenResponse, TokenServiceAttrs } from '../../types'

export const generateToken = async (
  key: string,
  params: Token,
  serviceAttrs: TokenServiceAttrs
): Promise<TokenResponse> => {
  const token = sign(params, key, {
    ...serviceAttrs,
    algorithm: 'RS256',
  })
  return {
    token,
  }
}
