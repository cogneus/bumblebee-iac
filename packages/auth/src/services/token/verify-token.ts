import { verify } from 'jsonwebtoken'
import { TokenServiceAttrs } from '../../types'

export const verifyToken = async <TToken>(
  key: string,
  token: string,
  { audience, issuer }: TokenServiceAttrs
): Promise<TToken> =>
  verify(token, key, {
    audience,
    issuer,
    algorithm: 'RS256',
  })
