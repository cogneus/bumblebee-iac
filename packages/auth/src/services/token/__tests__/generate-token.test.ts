import { Token } from 'bumblebee-common'
import fs from 'fs'
import path from 'path'
import { generateToken } from '../generate-token'
import { TokenServiceAttrs } from '../../../types'
import { verifyToken } from '../verify-token'

const file = path.join(__dirname, './', 'test-key.key')
const key = fs.readFileSync(file, 'utf8')

describe('Generate token tests', () => {
  beforeEach(() => {
    jest.clearAllMocks()
    jest.useFakeTimers()
    jest.setSystemTime(new Date('2023-01-01T00:00:00Z'))
  })

  const tokenParams: Token = {
    email: 'someone@test.com',
    givenName: 'Test',
    legalName: 'User',
    rights: [],
    subject: 'test',
    uid: 'id',
  }

  const serviceAttrs: TokenServiceAttrs = {
    audience: 'audience',
    issuer: 'issuer',
    expiresIn: '12h',
  }

  test('should generate and decode a token', async () => {
    const { token } = await generateToken(key, tokenParams, serviceAttrs)
    const decoded = await verifyToken(key, token, serviceAttrs)
    expect({ token, decoded }).toMatchSnapshot()
  })
})
