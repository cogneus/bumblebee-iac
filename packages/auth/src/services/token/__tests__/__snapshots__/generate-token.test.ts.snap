// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate token tests should generate and decode a token 1`] = `
{
  "decoded": {
    "aud": "audience",
    "email": "someone@test.com",
    "exp": 1672574400,
    "givenName": "Test",
    "iat": 1672531200,
    "iss": "issuer",
    "legalName": "User",
    "rights": [],
    "subject": "test",
    "uid": "id",
  },
  "token": "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6InNvbWVvbmVAdGVzdC5jb20iLCJnaXZlbk5hbWUiOiJUZXN0IiwibGVnYWxOYW1lIjoiVXNlciIsInJpZ2h0cyI6W10sInN1YmplY3QiOiJ0ZXN0IiwidWlkIjoiaWQiLCJpYXQiOjE2NzI1MzEyMDAsImV4cCI6MTY3MjU3NDQwMCwiYXVkIjoiYXVkaWVuY2UiLCJpc3MiOiJpc3N1ZXIifQ.Z6i_UNl5RpmEcT0pOhdOY-RQ7lCy0PwysiY_HM_t05T4r1DmiucLzCo_MHQC32rCzCpPQbvKZ24lRU-Ww1Uk-PlHyybvhwS4SPNub7V9M4HLzi4nu25zvNXrqBkPtjmkW33u18fhPwRO-xTcg2y1vLa1OpJDNkUiTEr1d8TTq0S48fMNgCXrXu2YfDFm6HkTyVVGG3uWtlbqrEIu1cyCOijxnAH-_U0YaPgp7y33u8_uQNgfslTRU3DquJiGnuQYWwr0J4U1D81F8sJMLXGj9kSJfmqIX63QNrp7s5HeKyIQ6inQdAk3JbJ9J8APkv3imjAVBpZcDLuF8-yzYXL8XA",
}
`;
